runtime/src/main.c

```diff
@@ -259,10 +259,9 @@ log_compiletime_config()
 	printf("\tArchitecture: %s\n", "x86_64");
 #endif
 
-	int ncores = NCORES;
-	printf("\tTotal Cores: %d\n", ncores);
-	int page_size = PAGE_SIZE;
-	printf("\tPage Size: %d\n", page_size);
+#ifdef PAGE_SIZE
```

Comment:
I think this might be required. If so, this probably shouldn't be in a conditional. You want also want an `#ifndef` that triggers a `#warn` if not set as well.